src/NosCore.GameObject/ComponentEntities/Extensions/AliveEntityExtension.cs

```diff
@@ -117,49 +117,54 @@ public static StPacket GenerateStatInfo(this IAliveEntity aliveEntity)
             };
         }
 
-        public static Task Move(this INonPlayableEntity nonPlayableEntity)
+        public static Task MoveAsync(this INonPlayableEntity nonPlayableEntity)
         {
             if (!nonPlayableEntity.IsAlive)
             {
                 return Task.CompletedTask;
             }
 
-            if (nonPlayableEntity.IsMoving && (nonPlayableEntity.Speed > 0))
+            if (!nonPlayableEntity.IsMoving || (nonPlayableEntity.Speed <= 0))
             {
-                var time = (SystemTime.Now() - nonPlayableEntity.LastMove).TotalMilliseconds;
+                return Task.CompletedTask;
+            }
 
-                if (time > RandomFactory.Instance.RandomNumber(400, 3200))
-                {
-                    var mapX = nonPlayableEntity.MapX;
-                    var mapY = nonPlayableEntity.MapY;
-                    if (nonPlayableEntity.MapInstance.Map.GetFreePosition(ref mapX, ref mapY,
-                        (byte)RandomFactory.Instance.RandomNumber(0, 3),
-                        (byte)RandomFactory.Instance.RandomNumber(0, 3)))
-                    {
-                        var distance = (int)Heuristic.Octile(Math.Abs(nonPlayableEntity.PositionX - mapX),
-                            Math.Abs(nonPlayableEntity.PositionY - mapY));
-                        var value = 1000d * distance / (2 * nonPlayableEntity.Speed);
-                        Observable.Timer(TimeSpan.FromMilliseconds(value))
-                            .Subscribe(
-                                _ =>
-                                {
-                                    nonPlayableEntity.PositionX = mapX;
-                                    nonPlayableEntity.PositionY = mapY;
-                                });
-
-                        nonPlayableEntity.LastMove = SystemTime.Now().AddMilliseconds(value);
-                        return nonPlayableEntity.MapInstance.SendPacket(
-                            nonPlayableEntity.GenerateMove(mapX, mapY));
-                    }
-                }
+            var time = (SystemTime.Now() - nonPlayableEntity.LastMove).TotalMilliseconds;
+
+            if (!(time > RandomFactory.Instance.RandomNumber(400, 3200)))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use the opposite operator ('<=') instead.](https://app.codacy.com/manual/NosCoreIO/NosCore/pullRequest?prid=5266673)